containers/daap-python-base/src/var/task/data_product_metadata.py

```diff
@@ -0,0 +1,117 @@
+import json
+import os
+import traceback
+
+import boto3
+import botocore
+from jsonschema import validate
+from jsonschema.exceptions import ValidationError
+
+from data_platform_logging import DataPlatformLogger
+from dataengineeringutils3.s3 import get_filepaths_from_s3_folder, read_json_from_s3
+
+s3_client = boto3.client("s3")
+
+
+# These can be moved to separate module for all paths when work starts on that
+def get_bucket_name():
+    return os.environ["BUCKET_NAME"]
+
+
+def get_data_product_metadata_path(data_product_name):
+    metadata_s3_path = os.path.join(
+        "s3://",
+        get_bucket_name(),
+        "metadata",
+        data_product_name,
+        "v1.0",
+        "metadata.json"
+    )
+    return metadata_s3_path
+
+
+def get_data_product_metadata_spec_path(version: str = None) -> str:
+    # if version is None we'll get the latest version
+    if version is None:
+        file_paths = get_filepaths_from_s3_folder(
+            os.path.join("s3://", get_bucket_name(), "data_product_metadata_spec")
+        )
+        versions = list(
+            {i for p in file_paths for i in p.split("/") if i.startswith("v")}
+        )
+        versions.sort()
```

Comment:
This is something we should have a test for once the unit test PR is merged. We might run into order problems here (depending on the version format) since the default is going to be lexicographic ordering, e.g. 

```
>>> sorted(['v1', 'v10', 'v2'])
['v1', 'v10', 'v2']
```

Reply:
good spot, yeah think this may need a slight re-working

Reply:
will address during unit test development